pkg/zson/docs/spec.md

```diff
@@ -227,32 +223,31 @@ if the parsed value does not match the indicated type in terms of number and
 sub-structure of value elements present and their interpretation as a valid
 string of the specified type.
 
-It is an error for a value to include a descriptor that has not previously
+It is an error for a value to include a descriptor that has not been previously
 defined by a descriptor directive.
 
 ### Character Escape Rules
 
-Any character in a value line may be escaped from the zson formatting rules
-using the javascript rules for escape syntax, i.e.,
-* \ddd for octal escapes,
-* \xdd for hex escapes,
-* \udddd for unicode escapes,
-* and the various single character escapes of javascript.
+Any character in a value line may be escaped from the ZSON formatting rules
+using the JavaScript rules for escape syntax, i.e.,
+* `\ddd` for octal escapes,
+* `\xdd` for hex escapes,
+* `\udddd` for unicode escapes,
+* and the various single character escapes of JavaScript.
 
-Sequences of binary data can be embedded in values using these escapes but its
+Sequences of binary data can be embedded in values using these escapes but it is
 a semantic error for arbitrary binary data to be carried by any types except
-string and bytes (see [Type Semantics](#type-semantics)).
+`string` and `bytes` (see [Type Semantics](#type-semantics)).
 
 These special characters must be escaped if they appear within a value:
 ```
 ; \n \\
 ```
-In addition, "-" must be escaped if the value is not the unset value but
-is a single ascii byte equal to "-".
+In addition, `-` must be escaped if representing the single ASCII byte equal to `-` as opposed to representing an unset value.
```

Comment:
This text is a little redundant with the closing sentence further up that talks about escaping unset. Should we maybe move that sentence down here, since this is the section on escaping?

Regardless of where it lives, I'd appreciate a confirmation that `\-` is the escaped representation we'd want to show. I kept it like that because it's what was there before. However, the JavaScript references I find online don't appear to treat `-` as a special character that needs escaping, so it doesn't strictly fall into the bucket of "the various single character escapes of JavaScript". If we were in the legacy zone I'd use the `\x` hex escaping, but we're in the regular zone. Is it our intent is to say that it's our equivalent of "a single character escape" as they appear in JavaScript, but that `-` happens to be a special character in ZSON, hence escaping it with just a `\` is fair game?

Reply:
@nwt volunteers to get back to this topic later.